src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -173,6 +180,50 @@ SimpleNameSyntax syntax
 			}
 		}
 
+		private static void AnalyzeConditionalImmutability( SyntaxNodeAnalysisContext ctx ) {
+			var syntax = ctx.Node;
+
+			// Get the symbol for the method
+			if( ctx.SemanticModel.GetDeclaredSymbol( ctx.Node ) is not IMethodSymbol symbol ) {
+				return;
+			}
+
+			// Exit if this somehow is not possible
+			// We don't care about arguments so throw them out
+			if( !GetTypeParamsAndArgs( symbol, out var typeParameters, out _ ) ) {
+				return;
+			}
+
+			// Retrieve the relevant parameter list
+			TypeParameterListSyntax paramListSyntax;
+			switch( syntax ) {
+				case MethodDeclarationSyntax:
+					paramListSyntax = ( (MethodDeclarationSyntax)syntax ).TypeParameterList;
+					break;
+				case LocalFunctionStatementSyntax:
+					paramListSyntax = ( (LocalFunctionStatementSyntax)syntax ).TypeParameterList;
+					break;

```

Comment:
Can avoid the extra cast

```suggestion
				case MethodDeclarationSyntax methodSyntax:
					paramListSyntax = methodSyntax.TypeParameterList;
					break;
				case LocalFunctionStatementSyntax localSyntax:
					paramListSyntax = localSyntax.TypeParameterList;
					break;
```


Though it would probably read even better as a switch expression:

```csharp
			// Retrieve the relevant parameter list
			TypeParameterListSyntax paramListSyntax = syntax switch {
				MethodDeclarationSyntax methodSyntax => methodSyntax.TypeParameterList,
				LocalFunctionStatementSyntax localSyntax => localSyntax.TypeParameterList,
				_ => throw new InvalidOperationException(),
			};
```